src/Annotation/AppEntityType.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Annotation;
+
+/**
+ * Defines an app Apigee Edge entity type annotation object.
+ *
+ * The annotation properties of entity types are found on
+ * \Drupal\apigee_edge\Entity\EdgeEntityType and are accessed using
+ * get/set methods defined in
+ * \Drupal\apigee_edge\Entity\EdgeEntityTypeInterface.
+ *
+ * @Annotation
+ */
+class AppEntityType extends EdgeEntityType {
+
+  /**
+   * {@inheritdoc}
+   *
+   * Use the generic app theme, if exists for all app entity type.
+   */
+  public $default_theme = 'app';
```

Comment:
https://github.com/apigee/apigee-devportal-kickstart-drupal/blob/8.x-1.x/modules/custom/apigee_kickstart_enhancement/src/Entity/AppViewBuilder.php#L37-L41 is deprecated, thanks for this.

Reply:
If we are going to do this here, then why not just add the `app.html.twig` template?  That would help keep the main template a bit cleaner.  I still maintain that having explicit templates would be the best and simplest to understand solution, despite the additional code. 

We are also missing theme hook suggestions here, which should be provided in this PR.

Reply:
With this, I intended to do something that Drupal core does too in the EntityViewBuilder: dynamically apply a theme on a render array if it exists, without explicitly requiring that theme to be defined. Template suggestions for this theme should be defined when the "app" theme gets defined.
If you call "apigee-entity.html.twig" as the main template, yes, I intend to keep it simple and does not add entity type specific additional classes to it as you suggested belove.